File: internal/app/discovery_api.go
package app

import (
	"context"

	"nens-k8s/internal/domain"
)

type DiscoveryAPI struct {
	discovery domain.APIDiscovery
	ctx       context.Context
}

func NewDiscoveryAPI(discovery domain.APIDiscovery) *DiscoveryAPI {
	return &DiscoveryAPI{discovery: discovery}
}

func (a *DiscoveryAPI) bind(ctx context.Context) {
	a.ctx = ctx
}

func (a *DiscoveryAPI) Resources(clusterID string) ([]domain.APIResource, error) {
	return a.discovery.Resources(a.ctx, clusterID)
}

func (a *DiscoveryAPI) Refresh(clusterID string) ([]domain.APIResource, error) {
	return a.discovery.Refresh(a.ctx, clusterID)
}

File: frontend/src/features/discovery/discovery.api.ts
import { Refresh, Resources } from '@bindings/go/app/DiscoveryAPI'
import { useClusters } from '@/features/clusters/cluster.store'
import { FIXTURE_RESOURCES } from './discovery.fixtures'
import type { ApiResource } from './discovery.types'

export async function listApiResources(clusterId: string): Promise<ApiResource[]> {
  if (useClusters.getState().offline) return FIXTURE_RESOURCES
  return (await Resources(clusterId)) as ApiResource[]
}

export async function refreshApiResources(clusterId: string): Promise<ApiResource[]> {
  if (useClusters.getState().offline) return FIXTURE_RESOURCES
  return (await Refresh(clusterId)) as ApiResource[]
}
